src/shared/resourcefetcher/utils.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { mkdirSync } from 'fs-extra'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { extensionSettingsPrefix } from '../constants'
+import { fileExists } from '../filesystemUtilities'
+import { getLogger } from '../logger'
+import { CompositeResourceFetcher } from './compositeResourceFetcher'
+import { FileResourceFetcher } from './fileResourceFetcher'
+import { HttpResourceFetcher } from './httpResourceFetcher'
+import { ResourceFetcher } from './resourcefetcher'
+
+/**
+ * Attempts to pull down a manifest and transform it into a URL to get an artifact
+ * @param manifestUrl URL to manifest
+ * @param urlTransform Function to parse manifest and output the artifact path
+ * @returns URL, or undefined if manifest was neither reachable nor transformable.
+ */
+export async function getManifestDetails(
+    manifestUrl: string,
+    urlTransform: (text: string) => { version: string; url: string } | undefined
+): Promise<{ version: string; url: string } | undefined> {
+    try {
+        const manifestFetcher = new HttpResourceFetcher(manifestUrl, { showUrl: true })
+        const manifest = await manifestFetcher.get()
```

Comment:
Try try{} should only wrap this get() call. Then the `urlTransform()` call can be wrapped in a different try. Then the caller can avoid a lot of boilerplate.

Reply:
Is the proposal to lift the `urlTransform` call out of this function to the caller? That would probably work, probably goes better with the spirit of the function name too.

Reply:
Mainly I am hoping to avoid the very verbose boilerplate at each of the call sites.